python/tests/unit/test_irauth.py

```diff
@@ -139,3 +139,63 @@ def test_irauth_grpcservice_version_default_v3():
 
     assert ext_auth_config['typed_config']['grpc_service']['envoy_grpc']['cluster_name'] == 'cluster_extauth_someservice_default'
     assert ext_auth_config['typed_config']['transport_api_version'] == 'V2'
+
+
+@pytest.mark.compilertest
+def test_irauth_includebody_v2():
+    yaml = """
+---
+apiVersion: getambassador.io/v3alpha1
+kind: AuthService
+metadata:
+  name:  mycoolauthservice
+  namespace: default
+spec:
+  auth_service: someservice
+  protocol_version: "v2"
+  proto: grpc
+  include_body:
+    allow_partial: true
+    body_raw_bytes: true
+    max_bytes: 4096
+"""
+    econf = _get_envoy_config(yaml, version='V2')
+
+    conf = econf.as_dict()
+    ext_auth_config = _get_ext_auth_config(conf)
+
+    assert ext_auth_config
+
+    request_body = ext_auth_config['typed_config']['with_request_body']
+    assert request_body['allow_partial_message'] == True
+    assert request_body['max_request_bytes'] == 4096
+    assert request_body['pack_as_bytes'] == True
+
+@pytest.mark.compilertest
+def test_irauth_includebody_version_v3():
+    yaml = """
+---
+apiVersion: getambassador.io/v3alpha1
+kind: AuthService
+metadata:
+  name:  mycoolauthservice
+  namespace: default
+spec:
+  auth_service: someservice
+  protocol_version: "v3"
+  proto: grpc
+  include_body:
+    allow_partial: true
+    body_raw_bytes: true
+    max_bytes: 4096
+"""
+    econf = _get_envoy_config(yaml, version='V3')
+
+    conf = econf.as_dict()
+    ext_auth_config = _get_ext_auth_config(conf)
+
+    assert ext_auth_config
+    request_body = ext_auth_config['typed_config']['with_request_body']
+    assert request_body['allow_partial_message'] == True
+    assert request_body['max_request_bytes'] == 4096
+    assert request_body['pack_as_bytes'] == True
```

Comment:
nittiest of nits: I don't like the red do-not-enter symbol that GH is showing here. Please add a trailing newline.

Relatedly, I recommend you configure your editor to obey the [`.editorconfig` file ](https://editorconfig.org/) (which says `insert_final_newline = true`).